cranelift/assembler-x64/meta/src/dsl/encoding.rs

```diff
@@ -99,6 +100,19 @@ pub struct Rex {
     pub digit: Option<u8>,
     /// The number of bits used as an immediate operand to the instruction.
     pub imm: Imm,
+    /// Used for `+rb`, `+rw`, `+rd`, and `+ro` instructions, which encode `reg`
+    /// bits in the opcode byte; if `Some`, this contains the expected bit width
+    /// of `reg`.
```

Comment:
Would it make sense to make this field an enum with `None`, `Rb`, `Rw`, `Rd`, `Ro` as options and a `.bits()` method? Not too important either way but that might make this a little more self-documenting/nicer.

Reply:
👍 I have another branch with that... abandoned because it was more lines. But, yeah, let me revive that since you mentioned it! I'll just add it to this PR and move the documentation around.